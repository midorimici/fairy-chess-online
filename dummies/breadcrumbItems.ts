export const items = [
  {
    name: 'Home',
    path: '/',
  },
  {
    name: 'Docs',
    path: '/docs',
  },
  {
    name: 'Breadcrumb',
    path: '/docs/breadcrumb',
  },
];
