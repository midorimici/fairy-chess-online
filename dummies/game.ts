export const game = {
  id: '1234567890abcdef',
  name: 'Game name',
  description:
    'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
  initialBoardImage: {
    url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
    width: 256,
    height: 256,
  },
};
