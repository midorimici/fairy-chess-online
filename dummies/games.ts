export const games = [
  {
    id: '1234567890abcdef',
    name: 'First game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
  {
    id: '1234567890abcde0',
    name: 'Second game',
    description:
      'Game desciption. This part may contains the description about the game or/and pieces used in the game.',
    initialBoardImage: {
      url: 'https://media.graphcms.com/zzxpvUPeT1yHQAO8vSrI',
      width: 256,
      height: 256,
    },
  },
];
