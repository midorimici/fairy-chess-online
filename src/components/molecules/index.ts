export { BreadcrumbNav } from './BreadcrumbNav';
export { Container } from './Container';
export { GameList } from './GameList';
