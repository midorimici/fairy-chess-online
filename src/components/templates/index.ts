export { Game } from './Game';
export { Games } from './Games';
export { Home } from './Home';
export { NewRoom } from './NewRoom';
