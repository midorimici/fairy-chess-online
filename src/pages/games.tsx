import type { GetStaticProps, NextPage } from 'next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import type { Namespace } from 'react-i18next';
import { Games } from '~/components/templates';
import { fetchGql } from '~/lib/graphcms';

type Props = {
  games: Game[];
};

const GamesPage: NextPage<Props> = ({ games }) => <Games games={games} />;

export default GamesPage;

const namespaces: Namespace = ['common'];

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  const data = await fetchGql<GamesResponse>(`
    {
      games {
        id
        name
        description
        initialBoardImage {
          url
          width
          height
        }
      }
    }
  `);

  return {
    props: {
      games: data.games,
      ...(await serverSideTranslations(locale ?? 'en', namespaces)),
    },
  };
};
