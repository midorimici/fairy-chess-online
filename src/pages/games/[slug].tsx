import type { GetStaticPaths, GetStaticProps, NextPage } from 'next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import type { Namespace } from 'react-i18next';
import { i18n } from 'next-i18next.config';
import { Game } from '~/components/templates';
import { fetchGql } from '~/lib/hygraph';

const GamePage: NextPage<GameResponse> = ({ game }) => <Game game={game} />;

export default GamePage;

const namespaces: Namespace = ['common'];

export const getStaticProps: GetStaticProps<{ game: Game }, { slug: string }> = async ({
  params,
  locale,
}) => {
  if (params === undefined) {
    return {
      notFound: true,
    };
  }

  const data = await fetchGql<GameResponse>(`
    {
      game(where: {slug: "${params.slug}"}) {
        slug
        name
        description
        initialBoardImage {
          url
          width
          height
        }
      }
    }
  `);

  return {
    props: {
      game: data.game,
      ...(await serverSideTranslations(locale ?? 'en', namespaces)),
    },
  };
};

export const getStaticPaths: GetStaticPaths<{ slug: string }> = async () => {
  const data = await fetchGql<{ games: { slug: string }[] }>(`
    {
      games {
        slug
      }
    }
  `);

  const paths = data.games
    .map((game: { slug: string }) =>
      i18n.locales.map((locale: string) => ({ params: { slug: game.slug }, locale }))
    )
    .flat();

  return {
    paths,
    fallback: false,
  };
};
