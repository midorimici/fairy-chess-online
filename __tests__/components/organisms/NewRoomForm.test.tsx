import { render } from '@testing-library/react';
import { games } from 'dummies/games';
import { NewRoomForm } from '~/components/organisms';

it('renders NewRoomForm unchanged', () => {
  const { asFragment } = render(<NewRoomForm games={games} />);
  expect(asFragment()).toMatchSnapshot();
});
