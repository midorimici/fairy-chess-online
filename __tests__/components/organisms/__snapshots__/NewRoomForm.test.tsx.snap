// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewRoomForm unchanged 1`] = `
<DocumentFragment>
  <form>
    <div
      class="css-0"
    >
      <div
        class="chakra-stack css-0"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r0:"
            id="field-:r0:-label"
          >
            game-type
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-select__wrapper css-0"
          >
            <select
              aria-describedby="field-:r0:-helptext"
              aria-required="true"
              class="chakra-select css-0"
              id="field-:r0:"
              required=""
            >
              <option
                selected=""
                value="first-game"
              >
                First game
              </option>
              <option
                value="second-game"
              >
                Second game
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-0"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <div
            class="chakra-form__helper-text css-0"
            id="field-:r0:-helptext"
          >
            Game desciption. This part may contains the description about the game or/and pieces used in the game.
          </div>
        </div>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-0"
            for="field-:r1:"
            id="field-:r1:-label"
          >
            room-visibility
            <span
              aria-hidden="true"
              class="chakra-form__required-indicator css-0"
              role="presentation"
            >
              *
            </span>
          </label>
          <div
            class="chakra-radio-group css-0"
            role="radiogroup"
          >
            <div
              class="chakra-stack css-0"
            >
              <label
                class="chakra-radio css-0"
                data-checked=""
              >
                <input
                  aria-required="true"
                  checked=""
                  class="chakra-radio__input"
                  id="radio-:r3:"
                  name="radio-:r2:"
                  required=""
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="public"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-0"
                  data-checked=""
                />
                <span
                  class="chakra-radio__label css-0"
                  data-checked=""
                >
                  public
                </span>
              </label>
              <label
                class="chakra-radio css-0"
              >
                <input
                  aria-required="true"
                  class="chakra-radio__input"
                  id="radio-:r4:"
                  name="radio-:r2:"
                  required=""
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="private"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-0"
                />
                <span
                  class="chakra-radio__label css-0"
                >
                  private
                </span>
              </label>
            </div>
          </div>
          <div
            class="chakra-form__helper-text css-0"
            id="field-:r1:-helptext"
          >
            public-description
          </div>
        </div>
        <button
          class="chakra-button css-0"
          type="submit"
        >
          create
        </button>
      </div>
      <a
        href="/games/first-game"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27256%27%20height=%27256%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            title="see-game-details"
          />
          <noscript />
        </span>
      </a>
    </div>
  </form>
</DocumentFragment>
`;
